lib/active_record/virtual_attributes.rb

```diff
@@ -75,12 +75,22 @@ def virtual_attribute?(name)
         )
       end
 
+      def virtual_attribute_alias?(name)
+        return false unless respond_to?(:attribute_alias?)
+
+        attribute_alias?(name) && virtual_attributes_to_define.keys.include?(name)
+      end
+
+      def virtual_attribute_alias_names
+        return [] unless respond_to?(:attribute_alias?)
+
+        virtual_attributes_to_define.keys.select {|name| attribute_alias?(name) }
+      end
+
       def virtual_attribute_names
-        if respond_to?(:column_names)
-          attribute_names - column_names
-        else
-          attribute_names
-        end
+        names = attribute_names | virtual_attribute_alias_names
+        names -= column_names if respond_to?(:column_names)
```

Comment:
@kbrock I believe this is the biggest question for this problem.  What exactly is an attribute and what is not.  It feels like it should be:

```
virtual_attributes_to_define.keys
```

Do you know why we're pulling from attribute_names instead of the list of virtual attributes to define?

Reply:
I think the end desire was to not have a virtual attributes list but rather just have it as a rails attribute with the added `arel` attribute